gobblin-utility/src/main/java/org/apache/gobblin/util/logs/LogCopier.java

```diff
@@ -536,14 +521,15 @@ private void copyChangesOfLogFile(Path srcFile, Path destFile) throws IOExceptio
 
       try (Closer closer = Closer.create()) {
         fsDataInputStream = closer.register(LogCopier.this.srcFs.open(srcFile));
-        // Seek to the the most recent position if it is available
-        LOGGER.debug(String.format("Reading log file %s from position %d", srcFile, this.currentPos));
-        fsDataInputStream.seek(this.currentPos);
         BufferedReader srcLogFileReader = closer.register(
             new BufferedReader(new InputStreamReader(fsDataInputStream, ConfigurationKeys.DEFAULT_CHARSET_ENCODING)));
 
-        FSDataOutputStream outputStream = LogCopier.this.destFs.exists(destFile)
-            ? LogCopier.this.destFs.append(destFile) : LogCopier.this.destFs.create(destFile);
+        //If file already exists, delete and re-copy the file.
+        if (LogCopier.this.destFs.exists(destFile)) {
+          LogCopier.this.destFs.delete(destFile, false);
+        }
+
+        FSDataOutputStream outputStream = LogCopier.this.destFs.create(destFile);
```

Comment:
This call seems creating an entry in namenode directly. Based my previous comment on race condition, if that happens, two threads are copying bytes into the same `destFile`. Then is that a problem if we see existed file and then delete-followed-by-recopy ?

Reply:
This should not happen. The set of files to copy in each iteration does not have duplicates. Further, each iteration finishes only when all copy tasks complete. So we should not have a situation where two or more threads to copy the same file.